Grangasamy/fixing sdk and add new events

grangasamy-msft opened the pull request:
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.


<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

grangasamy-msft committed:
Fixing inconsistencies on TypeSpec for SMSReceived event.

grangasamy-msft committed:
Added TypeSpec fix for SMSDeliveryReportReceived events

grangasamy-msft committed:
Bug fixes.

grangasamy-msft committed:
Fixing inconsistencies for Router events

grangasamy-msft committed:
Fixing Inconsistencies in Chat events

grangasamy-msft committed:
Added ChatAzureBotCommand event.

grangasamy-msft committed:
Added ChatTypingIndication event to SDK.

grangasamy-msft committed:
Fixing ChatTyping name from ChatTypingIndicator to ChatTypingIndicatorSent

grangasamy-msft committed:
Removing changes from 2018 jason file.

grangasamy-msft committed:
Bug fixes

grangasamy-msft committed:
Bug fix adding required to the swagger.

grangasamy-msft committed:
Bug fix. Changing the required for GeneratedSystemEvents.json 2018-01-01

grangasamy-msft committed:
Fixing the required property based on the decorator.

grangasamy-msft committed:
Updated event type to new names
ChatTypingIndicatorReceivedInThread and ChatAzureBotCommanReceivedInThread, and made changes accordingly.

grangasamy-msft committed:
Adding Call Started and Call Ended Events.

grangasamy-msft committed:
Added CallParticipant Added and Removed event.

grangasamy-msft committed:
Merge branch 'grangasamy/AddCallingEvents' of https://github.com/Azure/azure-rest-api-specs into grangasamy/FixingSDKAndAddNewEvents

grangasamy-msft committed:
Merge branch 'grangasamy/TypeSpecFixSMSEvents' of https://github.com/Azure/azure-rest-api-specs into grangasamy/FixingSDKAndAddNewEvents

grangasamy-msft committed:
Merge branch 'grangasamy/TypeSpecFixChatEvents' of https://github.com/Azure/azure-rest-api-specs into grangasamy/FixingSDKAndAddNewEvents

grangasamy-msft committed:
Merge branch 'grangasamy/TypeSpecFixRouterEvents' of https://github.com/Azure/azure-rest-api-specs into grangasamy/FixingSDKAndAddNewEvents

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[CallAutomation-Common](https://apiview.dev/Assemblies/Review/8bf97c81fbc949dca5725152648e7aa0?revisionId=89c3bfebba45428090fc4a2288728519)

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+
+/** Schema of common properties of all participant events */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsCallParticipantBaseEventData extends AcsCallingEventBaseProperties {
```
We shouldn't use Base in the event names.

grangasamy-msft replied:
Its not an event name Josh. It has the base properties for all CallParticipants.

JoshLove-msft replied:
It still should not have the word "Base" in the name.

grangasamy-msft replied:
Removed the "Base" in the name

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+
+/** Schema of common properties of all calling events */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsCallingEventBaseProperties {
```
We should remove Base.

grangasamy-msft replied:
It is not an event name Josh. It has the base / common properties for all CallingEvents.

JoshLove-msft replied:
We should not use "Base" in any of the new model names.

grangasamy-msft replied:
Removed "Base"

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  endedBy: CommunicationIdentifierModel;
+
+  /** Participant or Internal or Carrier or any future types */
+  type: string;
```
This should be an extensible enum.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  /** Participant or Internal or Carrier or any future types */
+  type: string;
+
+  /** MicrosoftInternal or the name of the carrier only 1 of the mri and the name field will be filled based on the type. */
```
What is the mri?

grangasamy-msft replied:
Its a resource identifier. We use that id to identify the participant in the call.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsCallingEventBaseProperties {
+  /** The communication identifier of the call startedBy */
+  startedBy?: CommunicationIdentifierModel;
```
Are these all optional properties?

grangasamy-msft replied:
Ya. I dont have a clear answer from the team yet so keeping it optional for now.

grangasamy-msft replied:
StartedBy changed to required.

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -911,6 +967,53 @@
+        "messageBody"
+      ]
+    },
+    "AcsChatTypingIndicatorSentEventData": {
```
The name doesn't match the tsp version.

grangasamy-msft replied:
Fixed it.

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/GeneratedSystemEvents.json:
```diff
@@ -229,6 +327,62 @@
         }
       ]
     },
+    "AcsCallParticipantAddedEventData": {
```
ReceivedInThreadEvent seems to be missing.

grangasamy-msft replied:
Hi Josh, 
The event name is CallParticipantAdded so we dont have ReceivedInThread for this.

JoshLove-msft replied:
I am saying that the event is missing from this file, yet it is in the tsp file. All the events from tsp should also be in here.

grangasamy-msft replied:
Added the missing event to tsp.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1003,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  displayName?: string;
+
+  /** The Id of the participant. */
+  participantId?: boolean;
```
This doesn't seem like the correct type.

grangasamy-msft committed:
Code optimize

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -1395,6 +1524,11 @@ union AcsRouterUpdatedWorkerProperty {
 
+/** CallEnded event Type properties that can be updated */
+union CallEndedTypeProperty {
+  string,
```
We should include the well known names here, e.g. Participant

JoshLove-msft replied:
Look at CommunicationCloudEnvironmentModel as an example.

JoshLove-msft replied:
This still isn't fixed.

grangasamy-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -889,6 +889,62 @@
+      "properties": {
+        "reason": {
+          "description": "The reason for the participant removed.",
+          "$ref": "#/definitions/AcsCallingEventReasonProperties"
```
Added only the reason. Didnt include removedBy since that the one we have confusion with. Let me know if I should removed both.

grangasamy-msft replied:
Removed it

grangasamy-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1771,6 +1933,107 @@
+          "type": "string"
+        },
+        "name": {
+          "description": "MicrosoftInternal or the name of the carrier only 1 of the mri and the name field will be filled based on the type.",
```
Fixed it.

grangasamy-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1771,6 +1933,107 @@
+          "type": "integer"
+        },
+        "subCode": {
+          "description": "Null for 1:1 calls and ACS group calls with group Id.",
```
Fixed it.

grangasamy-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1771,6 +1933,107 @@
+          "$ref": "#/definitions/AcsCallingEventGroupProperties"
+        },
+        "isTwoParty": {
+          "description": "The isTwoParty metadata",
```
Fixed it.

grangasamy-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -36,13 +36,65 @@ model AcsUserDisconnectedEventData {
+model AcsCallParticipantRemovedEventData
+  extends AcsCallParticipantBaseEventData {
+  /** The reason for the participant removed. */
+  reason?: AcsCallingEventReasonProperties;
```
Fixed it.

grangasamy-msft committed:
Resolving comments and bug

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1001,90 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  endedBy: CommunicationIdentifierModel;
+
+  /** The type of call ended by. */
+  type: CallEndedParticipantTypeProperty;
```
Do we need property at the end of this type? Also, it doesn't match the model definition name. What about `CallEndedParticipantKind`?

grangasamy-msft committed:
Fixing common json.

grangasamy-msft committed:
Fixing Calling events schema and Type enum.

grangasamy-msft committed:
Fixing common json

osaghaso commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -889,6 +889,58 @@
+      ],
+      "type": "object",
+      "properties": {
+      }
```
Why is this empty?

osaghaso replied:
you missing reason and removedBy? for participantRemovedEvent 
![image](https://github.com/user-attachments/assets/4e83cc68-8253-431b-bb47-f502f4689de8)

osaghaso commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1771,6 +1929,148 @@
+      "properties": {
+        "startedBy": {
+          "description": "The communication identifier of the call started by",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2023-11-15/common.json#/definitions/CommunicationIdentifierModel"
```
this should be -  "$ref": "#/definitions/AcsCallParticipantUserProperties"
![{2192BCC6-069F-49FF-9643-1769C2C9941D}](https://github.com/user-attachments/assets/7a65435b-8ad0-407a-8d2c-f79223ee4bb4)

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -36,13 +36,63 @@ model AcsUserDisconnectedEventData {
+/** Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallParticipantRemoved event. */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsCallParticipantRemovedEventData
+  extends AcsCallParticipantEventProperties {
```
you missing reason and removedBy? for participantRemovedEvent
![{F1238126-15DA-4B73-848C-25F636E1E530}](https://github.com/user-attachments/assets/5f11e8b3-13d4-495d-828a-7649555c9f2d)

osaghaso reviewed (commented):
please see my comments

grangasamy-msft committed:
Fixing startedBy property

grangasamy-msft committed:
Ran prettier - Fixing bug

pgrandhi commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1001,110 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  /** The type of call ended by. */
+  type: AcsCallEndedParticipantKind;
+
+  /** The name field will be filled based on the type. */
```
update comment, maybe "name of endedBy"

pgrandhi commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +1001,110 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  /** Reason for subcode for call ended event. */
+  subCode?: int;
+
+  /** Reason for the call ended event. */
```
not reason for call ended event, it is reason for ending the call

pgrandhi replied:
same for other comments above.

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -889,6 +889,58 @@
+      "properties": {
+        "endedBy": {
+          "description": "The communication identifier of the user who was disconnected",
+          "$ref": "#/definitions/AcsCallingEventEndedByProperties"
```
Should this be called AcsCallEndedByProperties? It is not CallingEventEndedBy

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -889,6 +889,58 @@
+        },
+        "reason": {
+          "description": "The reason for the call ended event.",
+          "$ref": "#/definitions/AcsCallingEventReasonProperties"
```
AcsCallEndedReasonProperties

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -911,6 +963,53 @@
+          "description": "The body of the chat message",
+          "type": "string"
+        },
+        "metadata": {
```
Is there metadata property in ChatAzureBotCommand event?

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/GeneratedSystemEvents.json:
```diff
@@ -254,6 +445,53 @@
         }
       ]
     },
+    "AcsChatAzureBotCommandReceivedInThreadEventData": {
```
shouldn't the schema look the same between chat typing indicator and chat azure bot command? They seem to be including different set of properties. one has metadata, other does not. Can you double check this?

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/chat_azure_bot_command_received_in_thread.json:
```diff
@@ -0,0 +1,36 @@
+    "composeTime": "2021-02-19T00:25:58.927Z",
+    "type": "Text",
+    "version": 1613694358927,
+    "recipientId": "8:acs:109f0644-b956-4cd9-87b1-71024f6e2f44_00000008-578d-7d05-83fe-084822000f6d",
```
. Can you double check? Is there recipient information in the payload?

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/chat_typing_indicator_received_in_thread.json:
```diff
@@ -0,0 +1,19 @@
+{
+  "source": "/subscriptions/{subscription-id}/resourceGroups/{group-name}/providers/Microsoft.Communication/communicationServices/{communication-services-resource-name}",
+  "subject": "thread/{thread-id}/sender/{rawId}/recipient/{rawId}",
```
Does this make sense to customer to include all these {} fields? Shouldn't this be plain text?

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/chat_azure_bot_command_received_in_thread.json:
```diff
@@ -0,0 +1,36 @@
+  "data": {
+    "messageBody": "Welcome to Azure Communication Services",
+    "messageId": "1613694358927",
+    "metadata": {
```
is there metadata property in chat azure bot command event?

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/chat_azure_bot_command_received_in_thread.json:
```diff
@@ -0,0 +1,36 @@
+      "key": "value",
+      "description": "A map of data associated with the message"
+    },
+    "senderId": "8:acs:109f0644-b956-4cd9-87b1-71024f6e2f44_00000008-578d-7caf-07fd-084822001724",
```
Is this copy/pasted from our internal code? If yes, it maybe exposing internal resource info. Generate a new Guid and copy here instead.

pgrandhi commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/call_started.json:
```diff
@@ -0,0 +1,28 @@
+    "startedBy": {
+      "communicationIdentifier": {
+        "kind": "communicationUser",
+        "rawId": "8:acs:22a04eb0-f822-483b-b9c9-75806ff3ff92_490a6903-af28-46cf-a2c7-c0ac73a7bbdb",
```
same here. do not copy from internal

pgrandhi commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -36,13 +36,63 @@ model AcsUserDisconnectedEventData {
+/** Schema of the Data property of an EventGridEvent for a Microsoft.Communication.ChatAzureBotCommandReceivedInThread event. */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
+model AcsChatAzureBotCommandReceivedInThreadEventData
+  extends AcsChatMessageEventInThreadBaseProperties {
```
Why is azureBotCommand extending from different model vs typing indicator event extending a different one? Shouldn't the schema be same for both?

pgrandhi replied:
what is the resolution here?

grangasamy-msft committed:
Resolving comments - Prema

pgrandhi commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -1037,7 +1037,7 @@ model AcsCallParticipantUserProperties {
 #suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "Maintain compatibility with existing models."
 model AcsCallingEventProperties {
-  /** The communication identifier of the call started user. */
+  /** The call participant who initialted the call. */
```
typo

grangasamy-msft committed:
Fixing typo

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_ended.json:
```diff
@@ -0,0 +1,45 @@
+      "name": "TestCallEndedBy"
+    },
+    "startedBy": {
+      "communicationIdentifier": {
```
this here is missing role. I know it's can not be present sometimes. we should always add the nullable values

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_ended.json:
```diff
@@ -0,0 +1,45 @@
+        }
+      }
+    },
+    "serverCallId": "Y3NVcmw",
```
"room" is missing.

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_participant_added.json:
```diff
@@ -0,0 +1,40 @@
+    "group": {
+      "id": "6d16b954-0531-4262-9cdd-c4b85da5c5d8"
+    },
+    "isTwoParty": null,
```
apply the same as the comments above to this file

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_participant_removed.json:
```diff
@@ -0,0 +1,45 @@
+      "communicationIdentifier": {
+        "kind": "communicationUser",
+        "rawId": "8:acs:d9ff4983-39e7-4420-8d54-f6354ed06ccd_7a668e62-08f4-4ac1-b29e-f01010f57475",
+        "communicationUser": {
```
update the properties "user" and "startedBy" to reflect the callparticipant model for the calling event examples
![image](https://github.com/user-attachments/assets/7e0fc314-178a-4985-bc36-7a44e67366f6)

grangasamy-msft committed:
Typo and comments update fix

grangasamy-msft committed:
Fixing the schema

grangasamy-msft committed:
Fixing the schema examples and adding the nullable

grangasamy-msft committed:
Schema fix

grangasamy-msft committed:
Fixing the naming to meaningful

grangasamy-msft committed:
Updating schemas and adding missing schema examples.

osaghaso commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2024-01-01/chat_typing_indicator_received_in_thread.json:
```diff
@@ -0,0 +1,19 @@
+        "id": "8:acs:109f0644-b956-4cd9-87b1-71024f6e2f44_00000008-578d-7caf-07fd-084822001724"
+      }
+    },
+    "senderDisplayName": "Jhon",
```
john? lol

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/GeneratedSystemEvents.json:
```diff
@@ -149,6 +149,145 @@
+          "type": "boolean"
+        }
+      },
+      "requried": [
```
```suggestion
      "required": [
```

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -958,7 +1062,11 @@
         }
-      }
+      },
+      "requried": [
```
```suggestion
      "required": [
```

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -985,7 +1093,11 @@
         }
-      }
+      },
+      "requried": [
```
```suggestion
      "required": [
```

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1001,7 +1113,10 @@
         }
-      }
+      },
+      "requried": [
```
```suggestion
      "required": [
```

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1017,7 +1132,10 @@
         }
-      }
+      },
+      "requried": [
```
```suggestion
      "required": [
```

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -958,7 +1062,11 @@
-      }
+      },
+      "requried": [
+        "editTime",
```
The required values shouldn't be changing for anything other than string or object type properties in 2018.

grangasamy-msft committed:
Fixing typo and changing enum back to string

grangasamy-msft committed:
Fixing the required parameter of Azure json file.

grangasamy-msft committed:
Fixing required typo

JoshLove-msft commented on specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:
```diff
@@ -1845,7 +2123,13 @@
+      "required": [
+        "messageId",
+        "type",
+        "version",
```
This cannot be made required.

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +997,110 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  startedBy: AcsCallParticipantProperties;
+
+  /** The call id of the server */
+  serverCallId?: string;
```
Should this be required?

pgrandhi commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -8,7 +8,7 @@ model AcsIncomingCallEventData {
   to: CommunicationIdentifierModel;
 
-  /** The communication identifier of the user who initiated the call. */
+  /** The communication identifier of the user who initialized the call. */
```
Please replace with initiated. It is not intialized

grangasamy-msft committed:
Fixing the typo

grangasamy-msft committed:
Fixing enum

grangasamy-msft committed:
Bug fixes.

grangasamy-msft committed:
Added required properties for calling events

grangasamy-msft committed:
Fixing the version property

grangasamy-msft committed:
Fixing the version.

grangasamy-msft committed:
Bug fix and system events

Commit:
Add JavaScript name customization

- AcsCallingEventProperties => AcsCallingEvent
- AcsCallParticipantEventProperties => AcsCallParticipantEvent

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_ended.json:
```diff
@@ -0,0 +1,49 @@
+    },
+    "callDurationInSeconds": 10,
+    "endedBy": {
+      "communicationidentifier": {
```
```suggestion
      "communicationIdentifier": {
```

richardpark-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/examples/2018-01-01/call_participant_removed.json:
```diff
@@ -0,0 +1,50 @@
+  "time": "2021-02-19T00:25:59.9436666Z",
+  "id": "02272459-badb-4e2e-b538-4cb8a2f71da6",
+  "data": {
+    "reason": {
```
This field isn't in the model for AcsCallParticipantRemovedEventData, or its parent objects. Is that intentional? I see the comment about it as resolved, but I'm not sure what the resolution was.

If the intention is to not expose it I'd recommend removing it from the example.

grangasamy-msft committed:
Fixing the call_ended and call_participant_removed schema examples.

samvaity commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -947,18 +997,110 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  communicationIdentifier: CommunicationIdentifierModel;
+
+  /** The type of call ended by. */
+  type: AcsCallEndedByKind;
```
rename to kind

JoshLove-msft replied:
We probably can't change the wire payload, btu we can apply a rename for all languages in https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/propertyNameOverride.tsp

samvaity commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -36,13 +36,59 @@ model AcsUserDisconnectedEventData {
+  reason?: AcsCallEndReasonProperties;
+
+  /** Duration of the call in seconds. */
+  callDurationInSeconds?: float64;
```
This should be `Duration`?

JoshLove-msft replied:
I think we will need to also use the `@encode` decorator so that it still gets encoded as a float.

samvaity committed:
ACS event renames (#34294)

mikeharder committed:
Merge branch 'main' into grangasamy/FixingSDKAndAddNewEvents

JoshLove-msft commented on specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Communication/AzureCommunicationServices.tsp:
```diff
@@ -963,18 +1014,110 @@ model AcsChatMessageEventBaseProperties extends AcsChatEventBaseProperties {
+  id?: string;
+}
+
+/** Schema of calling event endedby properties */
```
```suggestion
/** Schema of calling event ended by properties */
```

grangasamy-msft committed:
Pushing new auto gen swaggers and manual swaggers

grangasamy-msft committed:
Richard park changes.

grangasamy-msft committed:
Fixing the x-client-name for type property AcsCallEndedByProperties

grangasamy-msft committed:
Schema fix

grangasamy-msft committed:
Fixing chat typing example schema

l0lawrence committed:
Merge branch 'main' into grangasamy/FixingSDKAndAddNewEvents

l0lawrence committed:
add examples.json as @example and remove for validation warnings fix

l0lawrence committed:
typos/cspell

l0lawrence committed:
formatting

grangasamy-msft committed:
Adding format to integer type.

grangasamy-msft committed:
Merge branch 'grangasamy/FixingSDKAndAddNewEvents' of https://github.com/Azure/azure-rest-api-specs into grangasamy/FixingSDKAndAddNewEvents

grangasamy-msft committed:
Adding x-ms-examples for the new event schemas

grangasamy-msft committed:
Renamed snapshotCreated to Call/Chat Created x-ms-example.

grangasamy-msft committed:
Adding AcsCallParticipantRoleKind enum.

grangasamy-msft committed:
Fixing AcsCallParticipantRoleKind

grangasamy-msft committed:
Removing "x-ms-example" : "rolekind"

richardpark-msft committed:
Add rename for  AcsCallParticipantRoleKind to Go casing standards. (#34568)

Commit:
more js name overrides

Commit:
js name fix

grangasamy-msft committed:
Generating swaggers.

grangasamy-msft committed:
Fixing the tsp format for client and propertyNameOverride tsp.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.ApiCenter](https://apiview.dev/Assemblies/Review/3c9fafea50ce44d9a36a97c2e9751516?revisionId=006ec78c6b0345cf97bf65534befd080)
[sdk/messaging/eventgrid/azsystemevents](https://apiview.dev/Assemblies/Review/db71641e728c4b7ba62abe3efde3cf7e?revisionId=3a3bd85bd86042f88301bd4dd0fd4abe)
[Microsoft.EventGrid.SystemEvents](https://apiview.dev/Assemblies/Review/bf48b2982ff74cd796f5900a221af2d5?revisionId=5bc3029ebae74f36808e00abc0cb528f)
